openadapt/visualize1.py

```diff
@@ -0,0 +1,209 @@
+from pprint import pformat
+
+from loguru import logger
+from nicegui import ui
+from tqdm import tqdm
+
+from openadapt import config
+from openadapt.crud import get_latest_recording
+from openadapt.events import get_events
+from openadapt.utils import (
+    EMPTY,
+    configure_logging,
+    display_event,
+    image2utf8,
+    row2dict,
+    rows2dicts,
+    plot_performance,
+)
+
+SCRUB = config.SCRUB_ENABLED
+if SCRUB:
+    # too many warnings from scrubbing
+    __import__("warnings").filterwarnings("ignore", category=DeprecationWarning)
+    from openadapt import scrub
+
+
+LOG_LEVEL = "INFO"
+MAX_EVENTS = None
+MAX_TABLE_CHILDREN = 5
+PROCESS_EVENTS = True
+
+
+def create_tree(tree_dict, max_children=MAX_TABLE_CHILDREN):
+    tree_data = []
+    for key, value in tree_dict.items():
+        if value in EMPTY:
+            continue
+        node = {
+            "id": str(key)
+            + f"{(': '  + str(value)) if not isinstance(value, (dict, list)) else ''}"
+        }
+        if isinstance(value, dict):
+            node["children"] = create_tree(value)
+        elif isinstance(value, list):
+            if max_children is not None and len(value) > max_children:
+                node["children"] = create_tree(
+                    {i: v for i, v in enumerate(value[:max_children])}
+                )
+                node["children"].append({"id": "..."})
+            else:
+                node["children"] = create_tree({i: v for i, v in enumerate(value)})
+        tree_data.append(node)
+    return tree_data
+
+
+def set_tree_props(tree):
+    tree._props["dense"] = True
+    tree._props["no-transition"] = True
+    # tree._props["default-expand-all"] = True
```

Comment:
How about making this configurable?

Reply:
Yeah I'll use this to function to fetch the newvariables from openadapt.config 👍 just had it like this for testing